viewer/src/views/threejs/cad/materials.ts

```diff
@@ -22,9 +23,17 @@ export interface Materials {
   instancedMesh: THREE.ShaderMaterial;
   triangleMesh: THREE.ShaderMaterial;
   simple: THREE.ShaderMaterial;
+  // Data textures
+  colorDataTexture: THREE.DataTexture;
 }
 
 export function createMaterials(): Materials {
+  const pixelCount = 2048;
+  const colorCount = pixelCount * pixelCount;
+
+  const colors = new Uint8Array(4 * colorCount);
+  const colorDataTexture = new THREE.DataTexture(colors, pixelCount, pixelCount);
```

Comment:
How should we ensure that this function will be ready to create texture of the correct size to hold `maxTreeIndex` number of elements? Should it accept a CadMetadata-object (or whatever the current name is)?

Reply:
I think this is getting a bit too far ahead of the scope of this implementation. To create a texture of the correct size we need to also decide on/find solutions to:

- how and where we retrieve the information about the number of tree indices
- whether we can still create the materials and texture up-front or if we need to change the uniform later once data is received

I think it makes more sense to decide on how we will create a texture of the right size when we have all the necessary pieces rather than making too many assumptions about at this point.